src/libraries/Microsoft.Extensions.Http/src/Logging/LoggingHttpMessageHandler.cs

```diff
@@ -88,6 +89,8 @@ protected override HttpResponseMessage Send(HttpRequestMessage request, Cancella
         // Used in tests.
         internal static class Log
         {
+            private static readonly bool s_logQueryString = AppContext.TryGetSwitch("Microsoft.Extensions.Http.LogQueryString", out bool value) && value;
```

Comment:
The other name option I considered is `EnableQueryStringLogging`. Which one is better?

Reply:
It feels like `EnableQueryStringLogging` is better.

Reply:
I'd actually vote for using the same terminology that's _already used_ in dotnet/extensions logging (i.e. "_redaction_"), and also because it's a "_rollback_" switch instead of an "experimental feature" switch, I'd vote for saying "_disable_" instead of "enable":

`Microsoft.Extensions.Http.DisableUriRedaction`
or
`Microsoft.Extensions.Http.DisableUriQueryRedaction`

P.S.: We have at least 2 examples for similar "rollback" switches, and they also use "disable" wording: `System.Net.Security.DisableTlsResume` and `System.Net.Quic.DisableConfigurationCache`

P.P.S.: I also believe we've always added an _environmental variable_ alongside the AppContext switch (e.g. `DOTNET_SYSTEM_NET_SECURITY_DISABLETLSRESUME`) -- I think we should add one here as well.

Reply:
> We have at least 2 examples for similar "rollback" switches, and they also use "disable" wording: `System.Net.Security.DisableTlsResume` and `System.Net.Quic.DisableConfigurationCache`

From what I see, both switches have opposite semantics: a feature is enabled by default, and you need to explicitly assign a `true` to the switch value in-order to disable it. Same is true for `System.Net.SocketsHttpHandler.Http2FlowControl.DisableDynamicWindowSizing`.

In this case, query string logging would be disabled by default, so one would need to assign  `DisableUriQueryRedaction = false` in order to enable it. It might be just me, but it feels more confusing  than having an *Enable* switch that has to be assigned `true` in order to opt-in into the feature.

Examples for the logic I prefer: `System.ComponentModel.TypeConverter.EnableUnsafeBinaryFormatterInDesigntimeLicenseContextSerialization`,  `System.Diagnostics.DefaultActivityIdFormatIsHierarchial`, `DOTNET_SYSTEM_NET_SOCKETS_INLINE_COMPLETIONS`, `DOTNET_SYSTEM_RUNTIME_CACHING_TRACING`, `System.Resources.UseSystemResourceKeys` and many more.

Reply:
> From what I see, both switches have opposite semantics: a feature is enabled by default, and you need to explicitly assign a true to the switch value in-order to disable it.

But this is exactly what we have here, isn't it? Redaction = removing or substituting query string with "*". It _is_ enabled by default (with the change). If I want to "rollback" - to _stop redacting_ and see raw unredacted uris in my logs -- I say `DisableUriQueryRedaction = true`

Reply:
> `DisableUriQueryRedaction = true`

Ah sorry, that makes sense! Got lost a bit with the negations 😅